main.go

```diff
@@ -69,6 +81,10 @@ func main() {
 	flag.StringVar(&certPath, "ssl-cert-path", "", "Path to the TLS certificate (default: /opt/capsule-proxy/tls.crt)")
 	flag.StringVar(&keyPath, "ssl-key-path", "", "Path to the TLS certificate key (default: /opt/capsule-proxy/tls.key)")
 	flag.DurationVar(&rolebindingsResyncPeriod, "rolebindings-resync-period", 10*time.Hour, "Resync period for rolebindings reflector")
+	flag.Var(enumflag.NewSlice(&authTypes, "string", authTypesMap, enumflag.EnumCaseInsensitive), "auth-preferred-types",
```

Comment:
I would skip the `enumflag.EnumCaseInsensitive`, since it's a human input, we just need to document it properly.

Reply:
I've changed the enum flag to be case sensitive and have updated the flags docstring